firebase-installations/src/main/java/com/google/firebase/installations/Utils.java

```diff
@@ -15,13 +15,18 @@
 package com.google.firebase.installations;
 
 import android.text.TextUtils;
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.firebase.installations.local.PersistedInstallationEntry;
 import java.util.concurrent.TimeUnit;
 import java.util.regex.Pattern;
 
-/** Util methods used for {@link FirebaseInstallations} */
-class Utils {
+/**
+ * Util methods used for {@link FirebaseInstallations}.
+ *
+ * @hide
+ */
+public class Utils {
```

Comment:
Generally not a fan of classes that are called "Utils".
Seems like a dump for all methods that have no home, haha

Maybe we can have a time class that deals with all time related methods and has all time related constants, and call it "TimeSource" or "SystemTime" or "Clock" or something?
And another class that deals with validation and input checking, called "Validator" or something.

Even better would be if all these methods could find a home in the respective model classes (assuming that we have model classes).
Example: `#isAuthTokenExpired` goes into the `AuthToken` model class.

Reply:
I dont have a model class per se but I can clean up this class in another PR. Probably the one where I inject the Timesource .